packages/react-composites/src/composites/MeetingComposite/MeetingComposite.tsx

```diff
@@ -102,20 +104,44 @@ const MeetingScreen = (props: MeetingScreenProps): JSX.Element => {
 
   const [currentMeetingState, setCurrentMeetingState] = useState<CallState>();
   const [currentPage, setCurrentPage] = useState<MeetingCompositePage>();
-  meetingAdapter.onStateChange((newState) => {
-    setCurrentPage(newState.page);
-    setCurrentMeetingState(newState.meeting?.state);
-  });
-
+  const [unreadChatMessagesCount, setUnreadChatMessagesCount] = useState<number>(0);
   const [showChat, setShowChat] = useState(false);
   const [showPeople, setShowPeople] = useState(false);
 
+  useEffect(() => {
+    const updateMeetingPage = (newState): void => {
+      setCurrentPage(newState.page);
+      setCurrentMeetingState(newState.meeting?.state);
+    };
+    meetingAdapter.onStateChange(updateMeetingPage);
+    return () => {
+      meetingAdapter.offStateChange(updateMeetingPage);
+    };
+  }, [meetingAdapter]);
+  useEffect(() => {
```

Comment:
Is this a temporary implementation of how to count chat messages?

We try to avoid carrying business logic in `useEffect` blocks using the adapter API. That breaks a distinction we've tried to make between ACS business logic and UI. I can't think of a simple way to merge the inputs from ACS (incoming messages) and UI (last time chat pane was opened / whether the chat pane is open right now) in this case either.

If we take this approach, couple suggestions:
* Document why we can't use selectors to achieve what we want.
* Move this logic inside `NotificationIcon` component, to keep it encapsulated.

Reply:
The logic was out here incase we wanted to look at making the notification Icon a re-usable component with other buttons. So accessing the meeting state seemed best to do here at the composite level where we can subscribe to when a new message comes in and just pass a number around. I can leave a comment about why the composite holds this logic if you think thats best?

Reply:
Yeah using selectors is just going to be complicated but definitely possible -- we need the composite to store a local state of "seen messages", then have a selector filter `!state.messages.includes(compositeState.seenMessages)` then on chat open/close update the `compositeState.seenMessages`.
It'll conflict a little with the messageThread's internal implementation/interpretation of "unread messages" so some potential extra work to make sure those would marry up nicely.
But I think this is a good approach for a V1 implementation -- but also some great value if we can expose unreadMessages through the adapter API

Reply:
Note: I'm not complaining about the impl here (I can't think of a good selector based impl either).

But let's encapsulate this logic better.

Reply:
And the other reason this existed in the meetingcomposite file instead of the chatbutton / notification icon was that they don't have knowledge if the chat pane was opened/closed

Reply:
Let's get that knowledge about chat pane state prop-drilled down. I won't hold a hard line on this, but I do think that this logic needs to be encapsulated.

Reply:
will take a look at this!

Reply:
This is resolved in another PR.

Reply:
^ https://github.com/Azure/communication-ui-library/pull/1411